lib/json_api_client/middleware/status.rb

```diff
@@ -29,7 +31,7 @@ def handle_status(code, env)
         when 409
           raise Errors::Conflict, env
         when 400..499
-          # some other error
+          raise Errors::UnexpectedStatus.new(code, env[:url])
```

Comment:
That sounds right to me. Once the discussion with @Fivell reaches a conclusion, I'll update to do the more correct thing here.